src/apprt/gtk-ng/class/surface.zig

```diff
@@ -525,7 +525,76 @@ pub const Surface = extern struct {
 
     /// Ring the bell.
     pub fn ringBell(self: *Self) void {
-        // TODO: Audio feature
+        const window = ext.getAncestor(Window, self.as(gtk.Widget)) orelse {
```

Comment:
I wanted to do all this in the `Window` class by listening to the surface `ring-bell` signal. We should not be assuming any ancestor tree at all within `Surface`. The one place we do right now has a comment noting we should not do that (termio subprocess env).

Reply:
Fixed. One note is that I did not have the `Window` listen directly to the signal from the `Surface` because we need to know which `Tab` the surface is attached to. That's basically the function of the `bell` signal on the `Tab`, basically it just forwards the signal but with the `Tab` attached.